src/test/java/org/isf/pricesothers/test/Tests.java

```diff
@@ -127,6 +134,109 @@ public void testIoDeletePricesOthers() throws Exception {
+		foundPricesOthers.setDescription("Update");
+		assertThat(pricesOthersManager.updateOther(foundPricesOthers)).isTrue();
+		PricesOthers updatePricesOthers = priceOthersIoOperationRepository.findOne(id);
+		assertThat(updatePricesOthers.getDescription()).isEqualTo("Update");
+	}
+
+	@Test
+	public void testMgrNewPricesOther() throws Exception {
+		PricesOthers pricesOthers = testPricesOthers.setup(true);
+		assertThat(pricesOthersManager.newOther(pricesOthers)).isTrue();
+		_checkPricesOthersIntoDb(pricesOthers.getId());
+	}
+
+	@Test
+	public void testMgrDeletePricesOther() throws Exception {
+		int id = _setupTestPricesOthers(false);
+		PricesOthers foundPricesOthers = priceOthersIoOperationRepository.findOne(id);
+		assertThat(pricesOthersManager.deleteOther(foundPricesOthers)).isTrue();
+		assertThat(priceOthersIoOperationRepository.exists(id)).isFalse();
+	}
+
+	@Test
+	public void testMgrValidationCodeEmpty() throws Exception {
+		assertThatThrownBy(() ->
+		{
+			PricesOthers pricesOthers = testPricesOthers.setup(true);
+			pricesOthers.setCode("");
+			pricesOthersManager.newOther(pricesOthers);
+		})
+				.isInstanceOf(OHDataValidationException.class)
+				.has(
+						new Condition<Throwable>(
+								(e -> ((OHServiceException) e).getMessages().size() == 1), "Expecting single validation error")
+				);
+	}
+
+	@Test
+	public void testMgrValidationDescriptionEmpty() throws Exception {
+		assertThatThrownBy(() ->
+		{
+			PricesOthers pricesOthers = testPricesOthers.setup(true);
+			pricesOthers.setDescription("");
+			pricesOthersManager.newOther(pricesOthers);
+		})
+				.isInstanceOf(OHDataValidationException.class)
+				.has(
+						new Condition<Throwable>(
+								(e -> ((OHServiceException) e).getMessages().size() == 1), "Expecting single validation error")
+				);
+	}
+
+	@Test
+	public void testPricesOthersToString() throws Exception {
+		PricesOthers pricesOthers = new PricesOthers("TestCode", "TestDescription", true, false, false, true);
+		assertThat(pricesOthers).hasToString("TestDescription");
+	}
+
+	@Test
+	public void testPricesOthersEquals() throws Exception {
+		PricesOthers pricesOthers = new PricesOthers(-1, "TestCode", "TestDescription", true, false, false, true);
+
+		assertThat(pricesOthers.equals(pricesOthers)).isTrue();
+		assertThat(pricesOthers).isNotEqualTo(null);
+		assertThat(pricesOthers).isNotEqualTo("someString");
+
+		PricesOthers pricesOthers2 = testPricesOthers.setup(true);
+		pricesOthers2.setId(-99);
+		assertThat(pricesOthers).isNotEqualTo(pricesOthers2);
+
+		pricesOthers2.setId(-1);
+		assertThat(pricesOthers).isEqualTo(pricesOthers2);
+	}
+
+	@Test
+	public void testPricesOthersHashCode() throws Exception {
+		PricesOthers pricesOthers = testPricesOthers.setup(true);
+		pricesOthers.setId(1);
+		int hashCode = pricesOthers.hashCode();
+		assertThat(hashCode).isEqualTo(23 * 133 + 1);
+		// check computed value
+		assertThat(pricesOthers.hashCode()).isEqualTo(hashCode);
+	}
+
```

Comment:
testing is doing what it supposes to do, but why this `hashCode` method is only involving 1 single attribute from `PricesOthers` class?, there are 6 more attributes: `code`, `desc`, `opdInclude`, `ipdInclude`, `discharge`, and  `undefined`

cc @mwithi

Reply:
@emecas  I just test 'em 😄 and rarely do I question the implementations (except when they are *obviously* very wrong).

Reply:
Me too I didn't question that implementation at its time, but the same code is quite in every model's class at the first porting to JPA. I think we can fly over it for this PR and discuss it in another place.